hazelcast/db.py

```diff
@@ -0,0 +1,509 @@
+        raise NotSupportedError(f"Unknown type code: {code}")
+    return type
+
+
+_type_map = {
+    SqlColumnType.VARCHAR: Type.STRING,
+    SqlColumnType.BOOLEAN: Type.BOOLEAN,
+    SqlColumnType.TINYINT: Type.INTEGER,
+    SqlColumnType.SMALLINT: Type.INTEGER,
+    SqlColumnType.INTEGER: Type.INTEGER,
+    SqlColumnType.BIGINT: Type.INTEGER,
+    SqlColumnType.DECIMAL: Type.DECIMAL,
+    SqlColumnType.REAL: Type.FLOAT,
+    SqlColumnType.DOUBLE: Type.FLOAT,
+    SqlColumnType.DATE: Type.DATE,
+    SqlColumnType.TIME: Type.TIME,
+    SqlColumnType.TIMESTAMP: Type.DATETIME,
+    SqlColumnType.TIMESTAMP_WITH_TIME_ZONE: Type.DATETIME,
+    SqlColumnType.OBJECT: Type.OBJECT,
+    SqlColumnType.NULL: Type.NULL,
+    SqlColumnType.JSON: Type.JSON,
+}
+
+
+def _make_config(
+    config: Config = None,
+    *,
+    dsn="",
+    user: str = None,
+    password: str = None,
+    host: str = None,
+    port: int = None,
+    cluster_name: str = None,
+) -> Config:
+    kwargs_used = user or password or host or port or cluster_name
+    if config is not None:
+        if not isinstance(config, Config):
+            raise InterfaceError("config must be a hazelcast.Config object")
+        if dsn or kwargs_used:
+            raise InterfaceError("config argument cannot be used with keyword arguments")
+        return config
+    if dsn:
+        if kwargs_used:
+            raise InterfaceError("dsn argument cannot be used with other keyword arguments")
+        return _parse_dsn(dsn)
+    config = Config()
+    if not host:
+        host = "localhost"
+    if not port:
+        port = 5701
+    host = f"{host}:{port}"
+    config.cluster_members = [host]
+    if user is not None:
+        config.creds_username = user
+    if password is not None:
+        config.creds_password = password
+    if cluster_name is not None:
+        config.cluster_name = cluster_name
+    return config
+
+
+def _parse_dsn(dsn: str) -> Config:
+    r = urllib.parse.urlparse(dsn)
+    if r.scheme != "hz":
+        raise InterfaceError(f"Scheme must be hz, but it is: {r.scheme}")
+    cfg = Config()
+    host = "localhost"
+    port = 5701
+    if r.hostname:
+        host = r.hostname
+    if r.port:
+        port = r.port
+    cfg.cluster_members = [f"{host}:{port}"]
+    if r.username:
+        cfg.creds_username = r.username
+    if r.password:
+        cfg.creds_password = r.password
+    for k, v in urllib.parse.parse_qsl(r.query):
+        value: Any = v
+        if k in _parse_dsn_map:
+            attr_name, transform = _parse_dsn_map[k]
+            if transform:
+                try:
+                    value = transform(value)
+                except ValueError as e:
+                    raise InterfaceError from e
+            setattr(cfg, attr_name, value)
+        else:
+            raise InterfaceError(f"Unknown DSN attribute: {k}")
+    return cfg
+
+
+def _make_bool(v: str) -> bool:
+    if v == "true":
+        return True
+    if v == "false":
```

Comment:
Should we also allow capitalized `True` and `False`?

Reply:
This is just the DSN, so only `true` and `false` should be sufficient.